src/sidebar/services/annotation-activity.ts

```diff
@@ -43,12 +47,40 @@ export class AnnotationActivityService {
     };
 
     if (this._reportConfig.events.includes(eventType)) {
-      postMessageJsonRpc.notify(
-        this._rpc.targetFrame,
-        this._rpc.origin,
-        this._reportConfig.method,
-        [eventType, data],
-      );
+      this.notify(this._reportConfig.method, [eventType, data]);
     }
   }
+
+  /**
+   * Notify the embedder frame that the client has, or does not have,
+   * unsaved changes to annotations.
+   */
+  notifyUnsavedChanges(unsaved: boolean) {
+    // Note: Unlike activity reporting, the method name here is not configurable.
+    //
+    // The reason for having a configurable method name for activity reporting
+    // was because messages are sent via `window.postMessage` which is a shared
+    // channel that might be used for other purposes, hence configurable method
+    // names were added to avoid conflicts. The main consumer of this feature
+    // though is the Hypothesis LMS app, where we control all the JS code that
+    // runs. This method takes a simpler approach of using a fixed name and
+    // assuming it won't cause problems for the embedder frame.
+    //
+    // A better solution to this problem would be to use a dedicated
+    // `MessagePort` for communication.
+    this.notify('reportUnsavedChanges', [{ unsaved }]);
```

Comment:
Not using a configurable method name here is inconsistent with how the `reportActivity` method works, but it is simpler. I think the right way forwards would be to:

- Replace the configurable RPC method name in `reportActivity` with the fixed value "reportActivity", which is what the LMS app uses
- Change the communication channel to be a `MessagePort`, created by the client and posted to the LMS frontend via `window.postMessage`. With a dedicated channel for messaging, we don't have to worry about messages confusing other JS code that is running.